controllers/foodsController.js

```diff
@@ -78,4 +77,30 @@ const update = (req, res, next) => {
     })
 }
 
-module.exports = {index, show, create, update}
+const destroy = (req, res, next) => {
+  let foodId = req.params.id
+
+  let mealCount = Food.hasMeals(foodId)
+
+  Promise.all([mealCount]).then(mealCount => {
+    if (mealCount[0] === 0) {
+      Food.destroy(foodId)
+      .then(food => {
+        if(food.length === 0) {
```

Comment:
What is food here and what's the purpose of checking food.length. Is it to check for the event that a user attempts to delete a food that doesn't exist?

Reply:
Correct. The return from the database would be empty since the food doesn't exist.